src/components/AddHotel.jsx

```diff
@@ -0,0 +1,128 @@
+import React, { useState } from 'react';
+import { useSelector } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
+import { useDispatch } from 'react-redux';
+import { getAllHotelsAsync } from '../redux/hotel/hotel';
+
+const AddHotel = () => {
+  const [enteredName, setEnteredName] = useState('');
+  const [enteredDescription, setEnteredDescription] = useState('');
+  const [enteredImage, setEnteredImage] = useState(null);
+  const [enteredRating, setEnteredRating] = useState(0);
+  const [enteredOption, setEnteredOption] = useState('');
+
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+
+  const nameChangeHandler = (event) => {
+    setEnteredName(event.target.value);
+  };
+
+  const descriptionChangeHandler = (event) => {
+    setEnteredDescription(event.target.value);
+  };
+
+  const ratingChangeHandler = (event) => {
+    setEnteredRating(event.target.value);
+  };
+
+  const imageChangeHandler = (event) => {
+    setEnteredImage(event.target.files[0]);
+  };
+
+  const cityChangeHandler = (event) => {
+    setEnteredOption(event.target.value);
+  };
```

Comment:
- [ ] [Optional] To keep your code DRY, I strongly suggest you create a unique state and a unique `onChange` handler the code could be drastically reduced to:

``` javascript
const [newHotel, setNewHotel] = useState({
  name: '',
  description: '',
  picture: null,
  rating: 0,
  city: ''
}

const handleOnChange = (event) => {
  setNewHotel({ ...newHotel, 
  [event.target.name]: event.target.value
  })
}
```

For this implementation to work you have to assign an attribute `name` in your input with the same name as the key in the object:

``` javascript
<input
  name='name'
  type="text"
  value={enteredName}
  onChange={nameChangeHandler}
  autoComplete="off"
  className="appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
        />
```